Backport upstream commit:
http://git.savannah.gnu.org/cgit/gawk.git/commit/?id=982aa5c8

Fix to regex_internal.c from gawk list.
Resolves: #825693.
===================================================================
--- regex_internal.c.orig   2009-01-12 21:04:36.000000000 +0100
+++ regex_internal.c    2012-07-04 12:22:51.907889682 +0200
@@ -489,7 +489,7 @@ re_string_skip_chars (re_string_t *pstr,
        rawbuf_idx < new_raw_idx;)
     {
       int remain_len;
-      remain_len = pstr->len - rawbuf_idx;
+      remain_len = pstr->raw_len - rawbuf_idx;
       prev_st = pstr->cur_state;
       mbclen = __mbrtowc (&wc, (const char *) pstr->raw_mbs + rawbuf_idx,
 			  remain_len, &pstr->cur_state);
